### PR TITLE
chore: fix typescript annotations

### PR DESCRIPTION
**Pull request by stropho**
Just added typescript checking for some more files in my project and found a few typos.

### CODE REVIEW

File: lib/index.d.ts [hunk 1]
```diff
@@ -26,7 +27,7 @@ export interface IGlobPluginOptions {
      *   [acorn](https://www.npmjs.com/package/acorn) to check default exports.
      *   Intermediate files export both named exports and default exports.
      */
-    format?: IntermediateFileFormat,
+    format?: valueof<IntermediateFileFormat>,
```
Comment (kei-ito) [inline lib/index.d.ts]: Note: This fixes the problem ↓

note

I will rewrite this to `export type = 'import' | 'named' ...` later.

File: lib/index.d.ts [hunk 2]
```diff
@@ -45,7 +46,7 @@ export interface IGlobPluginOptions {
      * A function called before generating intermediate files.
      * You can add or remove files from sources.
      */
-    intercept?: (sources: Array<string>, importer: string, importee: string) => string,
+    intercept?: (sources: Array<string>, importer: string, importee: string) => Array<string>,
```
Comment (kei-ito) [inline lib/index.d.ts]: This is a my mistake.

File: lib/index.d.ts (hunk 1 above)
Comment (stropho) [reply lib/index.d.ts]: yup ` type = 'import' | 'named' ...` is probably more readable, but basically the same as `valueof<...>` .
Another solution would be to export the `IntermediateFileFormat` from your `js` file. Then, only those values could be used in the plugin when `format?: IntermediateFileFormat,` - a bit cumbersome but not entirely bad. Up to you :smile: 

 Would you mind releasing this little change soon ?

Comment (kei-ito) [reply lib/index.d.ts]: I've released v0.4.5 that includes the fixes.

**Pull request merged**